### DISCUSSION TITLE
Implementing Beamforming in Sionna

### DISCUSSION THREAD

**Discussion by anindha-dhar** [NVlabs/sionna#590] [Q&A] [answered]
I would like to use beamforming. Could you please tell me how to accomplish this with Sionna's assistance? I have three questions for that:

1. Does sionna have a built-in feature or something for beamforming? 
2. How can I use various beamforming techniques? such as beamforming based on codebooks?
3. Given that the receivers are mobile in some way, how can I put it into practice? 

Thank you.

**Comment by lllck** [comment]
> I would like to use beamforming. Could you please tell me how to accomplish this with Sionna's assistance? I have three questions for that:
> 
> 1. Does sionna have a built-in feature or something for beamforming?
> 2. How can I use various beamforming techniques? such as beamforming based on codebooks?
> 3. Given that the receivers are mobile in some way, how can I put it into practice?
> 
> Thank you.

Hi, have you solved these issues? I also want to achieve similar beamforming function.

**Comment by faycalaa** [comment, marked as answer]
Hi,

Is this the information you need? [MIMO Precoding API](https://nvlabs.github.io/sionna/phy/api/mimo.html#precoding)

**Reply by lllck** [reply]
Thanks, this helps me a lot
